public/js/logic.js

```diff
@@ -31,19 +34,40 @@ $("#container").on("click", "div.chest-wrapper", function(e){
   $(this).hide("fast", done(e));
 });
 
-function addChests(count) {
+function addChests(count, timer) {
+  let currentTime = Date.now();
+  let timeOut = timer * 60 * 1000 // Convert input in minutes to ms.
   for(let i = 1; i<=count; i++){
-    container.append(chest);
+    let chestID = 'ch-'+currentTime+'-'+i;
+    let currentChest = chest.replace('[timestamp]', chestID);
+    container.append(currentChest);
     chestCount++;
+
+    if (timer && timer != 0){
+      window['timer-'+chestID] = setTimeout(function() {deleteSpecificChest(chestID);}, timeOut);
```

Comment:
This still isn't quite working. I added 10 zero chests, then 2 1 minute chests. I then removed 10 chests manually (which includes the 2 1 minute chests), leaving 2 remaining chests. After 1 minute, the remaining 2 chests disappeared. I would expect the 2 minute chest timers to have been destroyed when their chests were removed.

0000000000
000000000022
22

You know, now that I visualize it, it may be an issue with the direction we are removing chests from.